pkg/ccl/backupccl/backupinfo/manifest_handling.go

```diff
@@ -529,6 +554,50 @@ func WriteBackupLock(
 	return cloud.WriteFile(ctx, defaultStore, lockFileName, bytes.NewReader([]byte("lock")))
 }
 
+// WriteSlimBackupManifestWithMetadataSSTs writes a "slim" version of manifest
+// to `exportStore`. This version currently has the alloc heavy `Files` repeated
+// field nil'ed out, and written to an accompanying SST instead.
+func WriteSlimBackupManifestWithMetadataSSTs(
+	ctx context.Context,
+	exportStore cloud.ExternalStorage,
+	encryption *jobspb.BackupEncryptionOptions,
+	kmsEnv cloud.KMSEnv,
+	manifest *backuppb.BackupManifest,
+) error {
+	if err := writeSlimBackupManifest(ctx, exportStore, backupbase.SlimBackupManifestName, encryption,
+		kmsEnv, manifest); err != nil {
+		return errors.Wrap(err, "failed to write the slim backup manifest")
+	}
+	return errors.Wrap(WriteBackupManifestFilesSST(ctx, exportStore, encryption, kmsEnv, manifest,
+		SlimManifestFileInfoPath), "failed to write backup manfiest files SST")
+}
+
+// WriteSlimBackupManifest compresses and writes a slim version of the
+// BackupManifest `desc` to `exportStore`.
+//
+// A slim backup manifest has the following repeated fields of the
+// BackupManifest proto nil'ed out:
+// - Files
+func writeSlimBackupManifest(
+	ctx context.Context,
+	exportStore cloud.ExternalStorage,
+	filename string,
+	encryption *jobspb.BackupEncryptionOptions,
+	kmsEnv cloud.KMSEnv,
+	manifest *backuppb.BackupManifest,
+) error {
+	slimManifest := *manifest
+	// We write a bogus file entry to ensure that no call site incorrectly uses
+	// the `Files` field from the slim manifest.
+	bogusFile := backuppb.BackupManifest_File{
+		Span: roachpb.Span{Key: roachpb.Key("bogusStart"), EndKey: roachpb.Key("bogusEnd")},
```

Comment:
do these bogus spans actually do what we want -- lead to things blowing up if we incorrectly use the manifest's list -- or do we need to make it the poison pill entry have a span of [min, max) to ensure it would be picked up by makeimportspans if we errantly use the wrong list and subsequently blow up when opened by restore data proc? 

relatedly: have we seen this fail if we e.g. don't set the bool?

Reply:
Added a test to show that we would fail during restore when opening the Files.Path.